src/iterator.c

```diff
@@ -1069,15 +1075,15 @@ static int filesystem_iterator_entry_cmp(const void *_a, const void *_b)
 	const filesystem_iterator_entry *a = (const filesystem_iterator_entry *)_a;
 	const filesystem_iterator_entry *b = (const filesystem_iterator_entry *)_b;
 
-	return git__strcmp(a->path, b->path);
+	return git__strcmp(a->basename, b->basename);
```

Comment:
This is a significant optimization. The current code sorts entries in a directory by their full paths. Full paths share long prefixes, which makes comparator slow, which makes sorting slow. TimSort, that libgit2 uses for sorting, performs insertion sort when there are fewer than 65 elements. Most directories have fewer than 65 entries. Insertion sort performs _a lot_ of comparisons, so it's very important to make comparator cheap. In gitstatusd I use an even cheaper (albeit more complex) comparator but only on Linux. This allows me to sort directory entries 7 times faster than in libgit2.

It's a good idea to add a comment here saying that `a->path` and `b->path` are guaranteed to be in the same directory, and maybe add an assert.